File: Part2_강의자료/Ch01_큐/문제별코드/15828_Router/15828.py
import sys
input = sys.stdin.readline
from collections import deque

N = int(input())
router = deque()
while True:
    cmd = int(input())
    if cmd > 0:
        if len(router) < N:
            router.append(cmd)
    elif cmd == 0 and router:
        router.popleft()
    else:
        break

if not router:
    print("empty")
else:
    print(*router)

File: Part2_강의자료/Ch01_큐/문제별코드/15828_Router/15828.cpp
#include <stdio.h>
#include <queue>
using namespace std;

int main()
{
    int N;

    scanf("%d", &N);
    queue<int> router;
    while (1)
    {
        int cmd;
        scanf("%d", &cmd);
        if (cmd > 0)
        {
            if (router.size() < N)
                router.push(cmd);
        }
        else if (cmd == 0)
            router.pop();
        else
            break;
    }

    if (router.empty())
        printf("empty");
    else
    {
        while (!router.empty())
        {
            printf("%d ", router.front());
            router.pop();
        }
        printf("\n");
    }
    return 0;
}
